flutter/news/lib/features/frontpage/data/datasource/articles_data_source.dart

```diff
@@ -1,6 +1,7 @@
+import 'package:news/core/result.dart';
 import 'package:news/features/frontpage/data/models/article_model.dart';
 
 abstract class ArticlesRemoteDataSource {
-  Future<List<ArticleModel>> getTopHeadlines();
-  Future<List<ArticleModel>> getEverythingAbout(String query);
+  Future<Result<List<ArticleModel>>> topHeadLines();
```

Comment:
The data sources will know enough in order to understand if the operation results in `success` or `failure`. If there are exceptions to catch, and then transform to `failure` it will happen there, on the bottom layer of the fetching mechanisms